tests/core/EncoderTests.cs

```diff
@@ -164,5 +164,74 @@ public void TestSymbolIcons()
                 Assert.IsTrue(encoder.Icons.ContainsKey(kind));
             }
         }
+
+        private string EncodeTestMagic(Documentation documentation)
+        {
+            var magic = new MagicSymbol
+            {
+                Name = "%test",
+                Documentation = documentation,
+                Kind = SymbolKind.Magic,
+                Execute = async (input, channel) => ExecutionResult.Aborted
+            };
+            return new MagicSymbolToHtmlResultEncoder().Encode(magic)?.Data ?? string.Empty;
+        }
+
+        [TestMethod]
+        public void TestMagicEncoder()
```

Comment:
NIT:
This is ok considering that it's a test method, but the set of Asserts is too verbose. Probably an auxiliary method that receives flags or a dictionary an calls Assert.IsFalse/IsTrue on each section accordingly might be briefer and more readable than the full set of asserts for each section. But this is merely a suggestion.

Reply:
Yes, this is a very valid point. I tend to prefer more verbose asserts because it can make it easier to debug when they fail. But in this case a helper method would save a lot of lines and wouldn't hurt the debuggability. I'll update it.